### PR TITLE
[#67] Group code snippets

### LINKED ISSUE #67: Group consecutive matches
For instance: https://codesearch.aelve.com/haskell/search?query=import&insensitive=on&sources=on. We can save a lot of space by grouping consecutive matches when their contexts overlap.

Before:

```
Graphics/Model/DirectX.hs

13 | import Control.Applicative
14 | import Data.Attoparsec.ByteString.Char8 as A
15 | import qualified Data.ByteString as B

Graphics/Model/DirectX.hs

13 | import Control.Applicative
14 | import Data.Attoparsec.ByteString.Char8 as A
15 | import qualified Data.ByteString as B
16 | import Data.Traversable

Graphics/Model/DirectX.hs

14 | import Data.Attoparsec.ByteString.Char8 as A
15 | import qualified Data.ByteString as B
16 | import Data.Traversable
17 | import Data.Word
```

After:

```
Graphics/Model/DirectX.hs

13 | import Control.Applicative
14 | import Data.Attoparsec.ByteString.Char8 as A
15 | import qualified Data.ByteString as B
16 | import Data.Traversable
17 | import Data.Word
```

### PR DESCRIPTION
**Pull request by ppressives**
Fixes #67

### CODE REVIEW

File: core/src/test/scala/codesearch/core/search/SnippetsGrouperSpec.scala [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package codesearch.core.search
+
+import cats.data.NonEmptyList
+import codesearch.core.config.SnippetConfig
+import codesearch.core.search.SnippetsGrouper.SnippetInfo
+import org.scalatest.{Matchers, WordSpec}
+
+class SnippetsGrouperSpec extends WordSpec with Matchers {
```
Comment (neongreen) [inline core/src/test/scala/codesearch/core/search/SnippetsGrouperSpec.scala]: Yay, tests, thanks!

Not sure this test will work properly, though – it might be better to have a folder with some fake Haskell code in the repo and use it for tests. But it's fine to do it in a separate PR.

**Comment by neongreen** [comment]
An example where two snippets overlap but have not been merged. Probably an off-by-one error:

image

**Comment by ppressives** [comment]
It was trouble in extraction rows for code snippet creation.
